result/result.py

```diff
@@ -1,29 +1,33 @@
 from typing import Callable, Generic, TypeVar, Union, Any, Optional, cast, overload
 
 
-E = TypeVar("E")
+T = TypeVar("T")  # Success type
+E = TypeVar("E")  # Error type
 F = TypeVar("F")
-T = TypeVar("T")
 U = TypeVar("U")
 
 
-class Result(Generic[E, T]):
+class Result(Generic[T, E]):
     """
     A simple `Result` type inspired by Rust.
 
     Not all methods (https://doc.rust-lang.org/std/result/enum.Result.html)
     have been implemented, only the ones that make sense in the Python context.
     """
-    def __init__(self, is_ok: bool, value: Union[E, T], force: bool = False) -> None:
+    def __init__(self, is_ok: bool, value: Union[T, E], force: bool = False) -> None:
         """Do not call this constructor, use the Ok or Err class methods instead.
 
         There are no type guarantees on the value if this is called directly.
 
         Args:
-            is_ok: If this represents an ok result
-            value: The value inside the result
-            force: Force creation of the object. This is false by default to prevent
+            is_ok:
+                If this represents an ok result
+            value:
+                The value inside the result
+            force:
+                Force creation of the object. This is false by default to prevent
                 accidentally creating instance of a Result in an unsafe way.
+
```

Comment:
Is it convention in python to add a trailing empty line?

Reply:
Not generally. I often do it to visually separate the docstring from the code.